src/wallet/wallet.h

```diff
@@ -648,6 +648,14 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
      */
     std::vector<CTxDestination> ListAddrBookAddresses(const std::optional<AddrBookFilter>& filter) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
 
+    /**
+     * Walk-through the address book entries.
+     * Stops when the provided 'ListAddrBookFunc' returns false.
+     */
+    using ListAddrBookFunc = std::function<void(const CTxDestination& dest, const std::string& label, const std::string& purpose, bool is_change)>;
+    void ForEachAddrBookEntry(const ListAddrBookFunc& func = [](const CTxDestination&, const std::string&, const std::string&, bool is_change){ return; }) const
```

Comment:
In 1db894772fb76d3469410683fc25f1c9721327cd "wallet: implement ForEachAddrBookEntry method"

I don't think it's necessary to have this default parameter for `func`. It isn't used and logically there is no reason why `ForEachAddrBookEntry` would ever be called without a specific `func`.

Reply:
yeah, good catch